app/frontend/src/features/auth/email/ConfirmChangeEmail.tsx

```diff
@@ -18,25 +20,39 @@ export default function ConfirmChangeEmail() {
   const { resetToken } = useParams<{ resetToken?: string }>();
 
   const {
+    data,
     error,
     isLoading,
     isSuccess,
-    mutate: completeChangeEmail,
-  } = useMutation<Empty, GrpcError, string>((resetToken) =>
-    service.account.completeChangeEmail(resetToken)
+    mutate: confirmChangeEmail,
+  } = useMutation<ConfirmChangeEmailRes.AsObject, GrpcError, string>(
+    (resetToken) => service.account.confirmChangeEmail(resetToken)
   );
 
   useEffect(() => {
     if (resetToken) {
-      completeChangeEmail(resetToken);
+      confirmChangeEmail(resetToken);
     }
-  }, [completeChangeEmail, resetToken]);
+  }, [confirmChangeEmail, resetToken]);
+
+  function successMsg(state: EmailConfirmationState) {
+    switch (state) {
+      case EmailConfirmationState.EMAIL_CONFIRMATION_STATE_SUCCESS:
+        return CHANGE_EMAIL_SUCCESS;
+      case EmailConfirmationState.EMAIL_CONFIRMATION_STATE_REQUIRES_CONFIRMATION_FROM_NEW_EMAIL:
+        return CHANGE_EMAIL_NEED_NEW;
+      case EmailConfirmationState.EMAIL_CONFIRMATION_STATE_REQUIRES_CONFIRMATION_FROM_OLD_EMAIL:
+        return CHANGE_EMAIL_NEED_OLD;
+      default:
+        return "Error";
+    }
+  }
 
   return isLoading ? (
     <Typography variant="body1">{CHANGE_EMAIL_PROGRESS}</Typography>
   ) : isSuccess ? (
     <>
-      <Alert severity="success">{CHANGE_EMAIL_SUCCESS}</Alert>
+      <Alert severity="success">{successMsg(data!.state)}</Alert>
```

Comment:
Should this not be `success` if it still requires confirming new/old email? Or should it not show the login button?

Reply:
Maybe hide the login link unless the state is `EMAIL_CONFIRMATION_STATE_SUCCESS`? Otherwise they'd still have to use their old email right?